src/components/Chat/ChatContent/Message/RoleSelector.tsx

```diff
@@ -21,6 +21,19 @@ const RoleSelector = React.memo(
     const currentChatIndex = useStore((state) => state.currentChatIndex);
 
     const [dropDown, setDropDown] = useState<boolean>(false);
+    const dropDownRef = React.useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+      const handleClickOutside = (event: any) => {
+        if (dropDownRef.current && !dropDownRef.current.contains(event.target)) {
+          setDropDown(false);
+        }
+      };
+      document.addEventListener('mousedown', handleClickOutside);
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    }, [dropDownRef]);
```

Comment:
Doing this would add an event listener to all RoleSelector on the page even if they are not toggled. This might negatively affect the performance if there are too many RoleSelector. I would suggest that the event listener to be added only when the the RoleSelector is "dropped down", and the event listener is removed when the RoleSelector is "not dropped down".

Reply:
I have added simple if-else statement inside useEffect to check if dropdown is open. If dropdown is open, we add the listener, otherwise we remove it.